Assets/Scripts/Tools/ExtendedRandom.cs

```diff
@@ -191,5 +179,13 @@ public int Next(int minValue, int maxValue)
         {
             return random.Next(minValue, maxValue);
         }
+
+        private Vector3 convertSphericalToCartesianCoords(float radius, float theta, float phi)
+        {
+            return new Vector3(
+                x: radius * Mathf.Sin(theta) * Mathf.Cos(phi),
+                y: radius * Mathf.Sin(theta) * Mathf.Sin(phi),
+                z: radius * Mathf.Cos(theta));
```

Comment:
Seems like theta and phi are reversed. Other than that everything looks good.
New to spherical coordinates but looked at chapter "Converting from Spherical Coordinates" in https://math.libretexts.org/Bookshelves/Calculus/Book%3A_Calculus_(OpenStax)/12%3A_Vectors_in_Space/12.07%3A_Cylindrical_and_Spherical_Coordinates

Reply:
It is correct as it is. The value which goes from 0->pi is, theta in this case is supposed to be where it is and the variable going from 0->2pi, phi in this case is supposed to be where it is. Though the normal notation is to have Theta going from 0->2pi, and phi going from 0->pi. I could change it if you really think it's important

Reply:
All good. As long as you know what you're doing